redbpf/src/lib.rs

```diff
@@ -550,10 +524,10 @@ impl XDP {
     pub fn attach_xdp(&mut self, interface: &str, flags: xdp::Flags) -> Result<()> {
         let fd = self.common.fd.ok_or(Error::ProgramNotLoaded)?;
         self.interfaces.push(interface.to_string());
-        let ciface = CString::new(interface).unwrap();
-        let res = unsafe { bpf_sys::bpf_attach_xdp(ciface.as_ptr(), fd, flags as u32) };
-
-        if res < 0 {
+        if let Err(e) = unsafe { attach_xdp(interface, fd, flags as u32) } {
+            if let Error::IO(oserr) = e {
+                eprintln!("error attaching xdp to interface {}: {}", interface, oserr);
```

Comment:
Is there another instance where we log to the console directly from the library? I think this can be a pretty controversial decision, and I'd generally prefer not introducing this pattern unless we already do it somewhere. A better option might be using `tracing`, which requires manual setup and can be filtered by the host app.

Reply:
Good point. I will fix this. Thank you :+1:

Reply:
Now `error!` of `tracing` crate is used instead of `eprintln!`